tests/integration/test_charm.py

```diff
@@ -135,3 +135,17 @@ async def test_openfga_relation(self, ops_test: OpsTest):
         assert ops_test.model.applications[APP_NAME].status == "active"
 
         await run_tctl_action(ops_test, namespace="integrations")
+
+    async def test_remove_server(self, ops_test: OpsTest):
+        """Admin charm goes to blocked state once relation with the server charm is removed."""
+        await ops_test.model.applications[SERVER_APP_NAME].destroy()
+        await ops_test.model.block_until(lambda: SERVER_APP_NAME not in ops_test.model.applications)
+
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME],
+            status="blocked",
+            raise_on_blocked=False,
+            timeout=300,
+        )
+
+        assert ops_test.model.applications[APP_NAME].units[0].workload_status == "blocked"
```

Comment:
Could it be worth writing a unit test that confirms the `BlockedStatus` message is `admin:temporal relation: database connections info not available` as it seems there are other `BlockedStatus` that could occur from the `self._setup_db_schemas(event)`

Reply:
Ah I see, sorry missed it as it was already in place :)